File: ios/SuiteBDE/Repositories/Analytics/AnalyticsRepository.swift
//
//  AnalyticsRepository.swift
//  BDE
//
//  Created by Nathan Fallet on 20/01/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import Firebase
import FirebaseAnalytics
import shared

class AnalyticsRepository: IAnalyticsRepository {
    
    func logEvent(name: AnalyticsEventName, params: [AnalyticsEventParameter : String]) {
        Analytics.logEvent(name.value, parameters: Dictionary(uniqueKeysWithValues: params.map { ($0.value, $1) }))
    }
    
    func setUserProperty(name: AnalyticsUserProperty, value: String?) {
        Analytics.setUserProperty(value, forName: name.value)
    }
    
}

File: android/src/main/kotlin/com/suitebde/repositories/analytics/AnalyticsRepository.kt
package com.suitebde.repositories.analytics

import com.google.firebase.Firebase
import com.google.firebase.analytics.analytics
import com.google.firebase.analytics.logEvent
import com.suitebde.models.analytics.AnalyticsEventName
import com.suitebde.models.analytics.AnalyticsEventParameter
import com.suitebde.models.analytics.AnalyticsUserProperty

class AnalyticsRepository : IAnalyticsRepository {

    override fun logEvent(name: AnalyticsEventName, params: Map<AnalyticsEventParameter, String>) =
        Firebase.analytics.logEvent(name.value) {
            params.forEach { (key, value) ->
                param(key.value, value)
            }
        }

    override fun setUserProperty(name: AnalyticsUserProperty, value: String?) =
        Firebase.analytics.setUserProperty(name.value, value)

}
